apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsLinkControl.tsx

```diff
@@ -0,0 +1,117 @@
+import type { ControlProps, RankedTester } from "@jsonforms/core"
+import {
+  Box,
+  FormControl,
+  HStack,
+  Icon,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Text,
+} from "@chakra-ui/react"
+import { and, isStringControl, rankWith, schemaMatches } from "@jsonforms/core"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+import { FormLabel, Input, Tab } from "@opengovsg/design-system-react"
+import { BiEnvelopeOpen, BiFile, BiFileBlank, BiLink } from "react-icons/bi"
+
+import { JSON_FORMS_RANKING } from "~/constants/formBuilder"
+
+export const jsonFormsLinkControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.LinkControl,
+  and(
+    isStringControl,
+    schemaMatches((schema) => schema.format === "link"),
+  ),
+)
+
+export function JsonFormsLinkControl({
+  data,
+  label,
+  handleChange,
+  path,
+  description,
+  required,
+}: ControlProps) {
+  const LINK_TYPES = [
+    {
+      icon: BiFileBlank,
+      label: "Page",
+    },
+    {
+      icon: BiLink,
+      label: "External link",
+    },
+    {
+      icon: BiFile,
+      label: "File",
+    },
+    {
+      icon: BiEnvelopeOpen,
+      label: "Email",
+    },
+  ]
+  const dataString = data && typeof data === "string" ? data : ""
+
+  return (
+    <Box py={2}>
+      <FormControl isRequired={required}>
+        <FormLabel description={description}>{label}</FormLabel>
+
+        <Tabs variant="group" onChange={() => handleChange(path, "")}>
```

Comment:
comment: i think tabs might be wrong way to go about this haha, i'd prefer a custom radio for semantic html. but since this is an internal facing application, might be ok

Reply:
Switched to radio in 950c8ed0b863961599692a2057cc26ee605d469c.